libsolidity/ast/AST.cpp

```diff
@@ -389,6 +391,68 @@ std::vector<std::pair<ASTPointer<IdentifierPath>, std::optional<Token>>> UsingFo
 	return ranges::zip_view(m_functionsOrLibrary, m_operators) | ranges::to<vector>;
 }
 
+void StructDefinition::insertEip712EncodedSubtypes(std::set<std::string>& subtypes) const
+{
+	for (size_t i = 0; i < m_members.size(); i++)
+	{
+		Declaration const* declaration = nullptr;
+
+		switch (m_members[i]->type()->category())
+		{
+			case Type::Category::Struct:
+				declaration = m_members[i]->type()->typeDefinition();
+				break;
+			case Type::Category::Array:
+				{
+					auto const* arrayType = dynamic_cast<ArrayType const*>(m_members[i]->type());
+					solAssert(!!arrayType);
+					if (auto finalyBaseType = dynamic_cast<StructType const*>(arrayType->finalBaseType(false)))
+					{
+						declaration = finalyBaseType->typeDefinition();
+					}
+				}
+				break;
+			default:
+				continue;
+		}
```

Comment:
Could be rewritten to just two `if`s, for example `if (auto arrayType = dynamic_cast<ArrayType const*>(type))` 

But I'm indifferent of getting this in.

Reply:
Ok